src/test/java/hudson/plugins/promoted_builds/integrations/jobdsl/PromotionsDslContextExtensionTest.java

```diff
@@ -41,4 +45,23 @@ public void testShouldGenerateTheDefindedComplexJob() throws Exception {
         // Then
         assertBuildStatusSuccess(scheduleBuild2);
     }
+
+    @Test
+    public void testShouldGenerateTheCopyArtifactsJob() throws Exception {
+        // Given
+        String dsl = FileUtils.readFileToString(new File("src/test/resources/copyartifacts-example-dsl.groovy"));
+        FreeStyleProject seedJob = createFreeStyleProject();
+        seedJob.getBuildersList().add(
+                new ExecuteDslScripts(new ExecuteDslScripts.ScriptLocation(Boolean.TRUE.toString(), null, dsl), false, RemovedJobAction.DELETE));
+        // When
+        QueueTaskFuture<FreeStyleBuild> scheduleBuild2 = seedJob.scheduleBuild2(0);
+        // Then (unstable b/c we aren't including the CopyArtifacts dependency)
+        assertBuildStatus(Result.UNSTABLE, scheduleBuild2.get());
```

Comment:
I'm not sure how to programmatically add plugins but if you point me to an example I will totally add that.  

However, I would offer that the 'unstable' is coming from the fact that the created job will not actually run due to the missing plugin but the test is about verifying the generated configuration.